File: doc/codes/csharp/examples/api/basics/geometry.cs
using System.Numerics;
using AUTD3;

namespace DocSamples.ApiBasicsGeometry;

internal static class Sample
{
    internal static void Run()
    {
        // ANCHOR: api
        new Geometry(new[] { new Autd3(
            Vector3.Zero,
            Quaternion.Identity
        ) });
        // ANCHOR_END: api

        var geometry = new Geometry(new[] { new Autd3(Vector3.Zero, Quaternion.Identity) });

        // ANCHOR: access
        var numDevices = geometry.NumDevices;
        var totalTransducers = geometry.NumTransducers;
        var arrayCenter = geometry.Center;

        for (var i = 0; i < geometry.NumDevices; i++)
        {
            var device = geometry[i];
        }

        var first = geometry[0];
        // ANCHOR_END: access

        var dev = geometry[0];
        // ANCHOR: device
        var idx = dev.Idx;
        var numTransducers = dev.NumTransducers;
        var center = dev.Center;
        var rotation = dev.Rotation;
        var x = dev.XDirection;
        var y = dev.YDirection;
        var axial = dev.AxialDirection;
        var positions = dev.Positions;
        var directions = dev.Directions;
        var pos0 = dev.Position(0);
        var dir0 = dev.Direction(0);
        // ANCHOR_END: device

        _ = (numDevices, totalTransducers, arrayCenter, first);
        _ = (idx, numTransducers, center, rotation, x, y, axial);
        _ = (positions, directions, pos0, dir0);
    }
}

File: doc/codes/rust/examples/api/basics/geometry.rs
use autd3_rs::geometry::{Autd3, Geometry, Point3, UnitQuaternion};

fn main() {
    // ANCHOR: api
    Geometry::new(vec![Autd3::new(
        Point3::origin(),
        UnitQuaternion::identity(),
    )]);
    // ANCHOR_END: api

    let geometry = Geometry::new(vec![Autd3::new(
        Point3::origin(),
        UnitQuaternion::identity(),
    )]);

    // ANCHOR: access
    let num_devices = geometry.num_devices();
    let total_transducers = geometry.num_transducers();
    let array_center = geometry.center();

    for device in &geometry {
        let _ = device;
    }

    let first = &geometry[0];
    // ANCHOR_END: access

    let device = &geometry[0];
    // ANCHOR: device
    let idx = device.idx();
    let num_transducers = device.num_transducers();
    let center = device.center();
    let rotation = device.rotation();
    let x = device.x_direction();
    let y = device.y_direction();
    let axial = device.axial_direction();
    let positions = device.positions();
    let directions = device.directions();
    let pos0 = device.position(0);
    let dir0 = device.direction(0);
    // ANCHOR_END: device

    let _ = (num_devices, total_transducers, array_center, first);
    let _ = (
        idx,
        num_transducers,
        center,
        rotation,
        x,
        y,
        axial,
        positions,
        directions,
        pos0,
        dir0,
    );
}
